pkg/utils/managedresources/managedresources.go

```diff
@@ -85,19 +167,30 @@ func DeleteManagedResource(ctx context.Context, client client.Client, namespace
 
 	if err := manager.
 		NewSecret(client).
-		WithNamespacedName(namespace, name).
+		WithNamespacedName(namespace, secretName).
 		Delete(ctx); err != nil {
-		return errors.Wrapf(err, "could not delete secret '%s/%s' of managed resource", namespace, name)
+		return errors.Wrapf(err, "could not delete secret '%s/%s' of managed resource", namespace, secretName)
 	}
 
 	return nil
 }
 
+var (
+	// DeleteForSeed is a function alias for deleteWithSecretNamePrefix.
+	DeleteForSeed = deleteWithSecretNamePrefix
+	// DeleteForShoot is a function alias for deleteWithSecretNamePrefix.
+	DeleteForShoot = deleteWithSecretNamePrefix
+)
+
+func deleteWithSecretNamePrefix(ctx context.Context, client client.Client, namespace string, name string) error {
+	return Delete(ctx, client, namespace, name, true)
+}
+
```

Comment:
Why do we need this? 
Can't the calling functions just directly call:
`Delete(ctx, client, namespace, name, true)` ?
And we delete the function `DeleteForSeed` and `DeleteForShoot`?

Reply:
Yes, you're right, we could do this of course. I've done it like this for two reasons:
1. To not touch too much of the other code (already using `DeleteForS{hoot,eed}` because the PR was already somewhat large.
1. This code already uses `CreateForS{hoot,eed}` which encapsulates the fact that the secret name should prefixed. For similarity reasons, there are also `DeleteForS{hoot,eed}` which encapsulates the same. If we now replace these functions by direct calls to `Delete` then the `secretNameWithPrefix` argument would be needed to be set to `true` explicitly - for all the places (while it's not needed when calling `CreateForS{hoot,eed}`. Honestly, I think this might lead to inconsistencies and would make the functions harder to use.

Reply:
ok, I get you logic. We can keep it then.